cmd/svcat/parameters/parameters.go

```diff
@@ -56,19 +60,38 @@ func ParseVariableAssignments(params []string) (map[string]interface{}, error) {
 		}
 		value := strings.TrimSpace(parts[1])
 
+		if strings.ContainsAny(variable, ".") {
+			newKeys = strings.Split(variable, ".")
+			variable = newKeys[0]
+			subKey = newKeys[1]
+			subMap[subKey] = value
+		}
+
 		storedValue, ok := variables[variable]
 		// Logic to add new value to map variables:
 		// if variable DNE: add pair to variables as variable:value
 		// if variable exists in form of variable:value, create array to hold old value&new value
 		// if variable exists in form variable:[some values], append new value to existing array
 		if !ok {
-			variables[variable] = value // if there is no key, add key&value as string
+			if len(subKey) == 0 {
+				variables[variable] = value // if there is no key, add key&value as string
+			} else {
+				variables[variable] = make(map[string]string)
+				variables[variable] = subMap
+			}
 		} else {
 			switch storedValType := storedValue.(type) {
 			case string:
 				variables[variable] = []string{storedValType, value}
 			case []string:
 				variables[variable] = append(storedValType, value)
+			case map[string]string:
+				if len(subKey) > 0 {
+					varsv, submapv := reflect.ValueOf(variables[variable]), reflect.ValueOf(subMap)
```

Comment:
I don't see what reflection is doing for us here? Is this different than doing the following?
```
for k, v := range sumMap {
   storedValType[k] = v
}
```
Ultimately, however, I don't think that we need to use a `subMap` at all since each loop through `params` is only going to have a single entry in `subMap`.

Reply:
hi @staebler thanks for your comments!! much appreciated.  

just to clarify that third case what is happening is:
storedValType[variable] = k[v] as opposed to storedValType[k]=v

so that storedValType[variable] can have values of ---> k[v], x[y], etc...

I went with the subMap, because the same k[v] map is used in two places (in if !ok and the 3rd case of the switch), so it seemed clearer to me how the k-v pair was being used that way. Will experiment and see what I can do with that this week. :)

Reply:
`storedValType` already is `variables[variable]`, so you would not want to then use `storedValType[variable]`.
Also, `k[v]` does not make sense since `k` is the key of an entry in the map and `v` is the value of the same entry. In other words, `subMap[k]` is `v`. `k` is a string not a map.

Reply:
@staebler sorry, I meant to say that the final value of the third switch statement should be 
variables[variable] that holds the map subMap[subKey]= subValue so that we end up with : 
map {variable:{subKey:subValue}}

Reply:
@staebler Thanks for your comments, I'll work on this switch statement more this week!